0.13: Light maps, irradiance volumes and reflection probes (#958)
bevyengine/bevy-website: content/news/2024-02-03-bevy-0.13/index.md

TODO.

## Basic baked global illumination

<div class="release-feature-authors">authors: @pcwalton</div>

> Computing lighting in real time is expensive!
> But for elements of a scene that never move (like rooms or terrain),

Reviewer: ```suggestion
Computing lighting in real time is expensive;
but for elements of a scene that never move (like rooms or terrain),
```

> we can get much prettier lighting and shadows for much cheaper by computing it ahead of time using **global illumination**,

Reviewer: ```suggestion
we can get prettier lighting and shadows for cheaper by computing it ahead of time using **global illumination**,
```
The adjectives by themselves are adequate for making the point concisely.

then storing the results in a "baked" form that never changes.

> Global illumination is a more sophisticated (and expensive) approach to lighting that often uses raytracing.

Reviewer: ```suggestion
Global illumination is a more sophisticated (and expensive) approach to lighting that often uses ray tracing.
```
The term `"Ray tracing"` is two separate words.

producing more realistic effects through the inclusion of indirect light.

**Lightmaps**, textures that store pre-computed results of this global illumination, have been a mainstay of real-time graphics for decades.
The [lightmap PR] adds preliminary support for them to Bevy.
Like the [lightmaps example] shows, just load in your baked lightmap image, and then insert a [`Lightmap`] component on the corresponding mesh.

![A simple scene rendered using baked light maps in Bevy 0.13](light_map.png)

Following up on this work, the [irradiance volumes PR] added support for a second form of baked global illumination.
**Irradiance volumes** (or voxel global illumination) is a technique used for approximating indirect light by first dividing
a scene into cubes (voxels), then sampling the amount of light present at the center of each of those voxels.
This light is then added to objects within that space as they move through it, changing the ambient light level on those objects appropriately.
As the PR helpfully explains, we've chosen to use the ambient cubes algorithm for this, based on Half Life 2.

> This allows use to match Blender's [Eevee renderer], giving users a simple and free path to creating nice looking irradiance volumes for their own scenes.

Reviewer 2: ```suggestion
This allows us to match Blender's [Eevee renderer], giving users a simple and free path to creating nice-looking irradiance volumes for their own scenes.
```

Note that this work does not include any Bevy-native way to precompute global illumination!
Instead for now this should be computed externally in a tool like [The Lightmapper], and then loaded into Bevy.

[lightmap PR]: https://github.com/bevyengine/bevy/pull/10231
[`Lightmap`]: https://dev-docs.bevyengine.org/bevy/pbr/struct.Lightmap.html
[lightmaps example]: https://github.com/bevyengine/bevy/blob/main/examples/3d/lightmaps.rs
[irradiance volumes PR]: https://github.com/bevyengine/bevy/pull/10268
[The Lightmapper]: https://github.com/Naxela/The_Lightmapper
[Eevee renderer]: https://docs.blender.org/manual/en/latest/render/eevee/index.html

## Minimal Reflection Probes

<div class="release-feature-authors">authors: @pcwalton</div>

> **Environment maps** are 2D textures used to simulate lighting, reflection and skyboxes in a 3D scene.

Reviewer: ```suggestion
**Environment maps** are 2D textures used to simulate lighting, reflection, and skyboxes in a 3D scene.
```
Oxford comma nitpick.

**Reflection probes** generalize environment maps to allow for multiple environment maps in the same scene, each of which has its own axis-aligned bounding box.
This is a standard feature of physically-based renderers and was inspired by the [corresponding feature in Blender's Eevee renderer].

> The [reflection probes PR], we've added basic support for these, laying the groundwork for pretty, high-performance reflections in Bevy games.

Reviewer: ```suggestion
In the [reflection probes PR], we've added basic support for these, laying the groundwork for pretty, high-performance reflections in Bevy games.
```
